packages/core/src/lib/template-engine/templateEngine.ts

```diff
@@ -65,6 +66,19 @@ export class TemplateEngine {
     data: T,
     pagination?: Pagination
   ): Promise<any> {
-    return this.compiler.execute(templateName, data, pagination);
+    const others = omit(data, '_prepared');
+    const prepared: PreparedQueryParams | undefined = get(data, '_prepared');
+    // wrap to context object
+    return this.compiler.execute(
+      templateName,
+      {
+        context: {
+          ...others,
+          ['params']: prepared?.identifiers || {},
+          ['_paramBinds']: prepared?.binds || {},
+        },
+      },
+      pagination
+    );
```

Comment:
Since we don't want users to access _paramBinds via templates, how about moving it out of `context` properties?

```suggestion
    return this.compiler.execute(
      templateName,
      {
        context: {
          ...others,
          ['params']: prepared?.identifiers || {},
        },
        ['_paramBinds']: prepared?.binds || {},
      },
      pagination
    );
```